Flaky RemoteReporter test

https://travis-ci.org/uber/jaeger-client-java/jobs/232521027#L863

```
com.uber.jaeger.reporters.RemoteReporterTest > testFlushUpdatesQueueLength FAILED
    java.lang.AssertionError: expected:<0> but was:<1>
        at org.junit.Assert.fail(Assert.java:88)
        at org.junit.Assert.failNotEquals(Assert.java:834)
        at org.junit.Assert.assertEquals(Assert.java:645)
        at org.junit.Assert.assertEquals(Assert.java:631)
        at com.uber.jaeger.reporters.RemoteReporterTest.testFlushUpdatesQueueLength(RemoteReporterTest.java:213)
```

--- a/jaeger-core/src/main/java/com/uber/jaeger/reporters/RemoteReporter.java
+++ b/jaeger-core/src/main/java/com/uber/jaeger/reporters/RemoteReporter.java
@@ -73,7 +73,7 @@ public class RemoteReporter implements Reporter {
             flush();
           }
         },
-        0,
+        flushInterval,
         flushInterval);
   }
 

--- a/jaeger-core/src/test/java/com/uber/jaeger/reporters/RemoteReporterTest.java
+++ b/jaeger-core/src/test/java/com/uber/jaeger/reporters/RemoteReporterTest.java
@@ -214,7 +214,10 @@ public class RemoteReporterTest {
     // change sender to blocking mode
     sender.permitAppend(0);
     RemoteReporter remoteReporter = (RemoteReporter) reporter;
-    remoteReporter.flush();
+
+    await("flush() execution via TimerTask").atMost(5, TimeUnit.SECONDS).until(
+        () -> metricsReporter.gauges.get("jaeger.reporter-queue") != null
+    );
 
     for (int i = 0; i < 10; i++) {
       reporter.report(newSpan());

--- a/jaeger-core/src/test/java/com/uber/jaeger/reporters/RemoteReporterTest.java
+++ b/jaeger-core/src/test/java/com/uber/jaeger/reporters/RemoteReporterTest.java
@@ -213,7 +213,6 @@ public class RemoteReporterTest {
   public void testFlushUpdatesQueueLength() throws Exception {
     // change sender to blocking mode
     sender.permitAppend(0);
-    RemoteReporter remoteReporter = (RemoteReporter) reporter;
 
     await("flush() execution via TimerTask").atMost(5, TimeUnit.SECONDS).until(
         () -> metricsReporter.gauges.get("jaeger.reporter-queue") != null
@@ -225,6 +224,7 @@ public class RemoteReporterTest {
 
     assertEquals(0, metricsReporter.gauges.get("jaeger.reporter-queue").longValue());
 
+    RemoteReporter remoteReporter = (RemoteReporter) reporter;
     remoteReporter.flush();
 
     assertTrue(metricsReporter.gauges.get("jaeger.reporter-queue") > 5);

--- a/jaeger-core/src/test/java/com/uber/jaeger/reporters/RemoteReporterTest.java
+++ b/jaeger-core/src/test/java/com/uber/jaeger/reporters/RemoteReporterTest.java
@@ -27,6 +27,7 @@ import static org.hamcrest.CoreMatchers.anyOf;
 import static org.hamcrest.CoreMatchers.equalTo;
 import static org.junit.Assert.assertEquals;
 import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertNull;
 import static org.junit.Assert.assertThat;
 import static org.junit.Assert.assertTrue;
 
@@ -211,23 +212,25 @@ public class RemoteReporterTest {
 
   @Test
   public void testFlushUpdatesQueueLength() throws Exception {
+    int neverFlushInterval = Integer.MAX_VALUE;
+    reporter = new RemoteReporter(sender, neverFlushInterval, maxQueueSize, metrics);
+    tracer = new Tracer.Builder("test-remote-reporter", reporter, new ConstSampler(true))
+        .withStatsReporter(metricsReporter)
+        .build();
+
     // change sender to blocking mode
     sender.permitAppend(0);
 
-    await("flush() execution via TimerTask").atMost(5, TimeUnit.SECONDS).until(
-        () -> metricsReporter.gauges.get("jaeger.reporter-queue") != null
-    );
-
-    for (int i = 0; i < 10; i++) {
+    for (int i = 0; i < 3; i++) {
       reporter.report(newSpan());
     }
 
-    assertEquals(0, metricsReporter.gauges.get("jaeger.reporter-queue").longValue());
+    assertNull(metricsReporter.gauges.get("jaeger.reporter-queue"));
 
     RemoteReporter remoteReporter = (RemoteReporter) reporter;
     remoteReporter.flush();
 
-    assertTrue(metricsReporter.gauges.get("jaeger.reporter-queue") > 5);
+    assertTrue(metricsReporter.gauges.get("jaeger.reporter-queue") > 0);
   }
 
   private Span newSpan() {

